src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -8481,17 +8479,26 @@ MethodTableBuilder::HandleExplicitLayout(
         // "i" indexes all fields, valueClassCacheIndex indexes non-static fields only. Don't get them confused!
         valueClassCacheIndex++;
 
-        if (CorTypeInfo::IsObjRef(pFD->GetFieldType()))
+        CorElementType type = pFD->GetFieldType();
+        if (CorTypeInfo::IsObjRef(type) || CorTypeInfo::IsByRef(type))
         {
             // Check that the ref offset is pointer aligned
             if ((pFD->GetOffset_NoLogging() & ((ULONG)TARGET_POINTER_SIZE - 1)) != 0)
             {
                 badOffset = pFD->GetOffset_NoLogging();
                 fieldTrust.SetTrust(ExplicitFieldTrust::kNone);
 
-                // If we got here, OREF field was not pointer aligned. THROW.
+                // If we got here, OREF or BYREF field was not pointer aligned. THROW.
                 break;
             }
+
+            // Other than alignment, a byref can be treated like a native pointer.
+            if (CorTypeInfo::IsByRef(type))
+                type = ELEMENT_TYPE_I;
```

Comment:
Is this used? From what I see below the only place that reads `type` is `if (CorTypeInfo::IsObjRef(type))` and that one will do the right thing no matter whether we bash this to `ELEMENT_TYPE_I`.

Reply:
This value isn't used directly, but I was attempting to indicate how it is being interpreted. I was attempting to demonstrate the behavior I am trying to capture.

Reply:
In this review, it gave me a pause because I saw that the codepath we're ending up this value is not used and we go back to the field and asking question of the field, ignoring the bashed `type`. It made me question whether that's not a bug. But maybe that's just the way I read it.

Reply:
That is fair. I'll change it.